### PR TITLE
Update to support pandera v0.21.0

### PR DESCRIPTION
**Pull request by sssangha**
Following #44 , I've proposed a custom patch for handling `GeometryDtype` with the `pandera` `v0.21.0` update

### CODE REVIEW

File: src/tile_mate/tile_model.py [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import geopandas as gpd
 from pandera import Column, DataFrameSchema
 
+# Add support for GeometryDtype
+pandas_engine.Engine.dtype = staticmethod(
+    lambda dtype: dtype if dtype == gpd.array.GeometryDtype else None
+)
+
```
Comment (cmarshak) [inline src/tile_mate/tile_model.py:8]: Why are you writing this yourself? I don't understand this function - it returns True or `None`? 

Why not what they recommend: https://pandera.readthedocs.io/en/stable/geopandas.html

**Comment by cmarshak** [comment]
I will have to look at this a little later - I am confused as to why we need the patch when pandera supports this on its docs. It might be worth fixing with pandera - and providing a temporary cap until its fixed.

File: src/tile_mate/tile_model.py (hunk 1 above)
Comment (cmarshak) [reply src/tile_mate/tile_model.py:8]: Maybe it's just about using `pa.Column("geometry")` and pandera takes care of all that for you. That's what I would recommend.

I am concerned we are loosing out on the validation with this.

Comment (sssangha) [reply src/tile_mate/tile_model.py]: @cmarshak good call. Yeah, I had no particular reason going about it with my original proposed changes, but I've now modified it by using pa.Column("geometry") directly to preserve validation.

Looks like it's also backwards compatible with the older `pandera` version.

**Review by cmarshak — APPROVED** [review]

**Pull request merged**